docs/deploy-full-rp-service-in-dev.md

```diff
@@ -0,0 +1,131 @@
+    > __NOTE:__ This will fail on the first attempt to run due to certificate and container mirroring requirements.
+
+1. Update the certificates in keyvault
+    ```bash
+    az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name rp-mdm \
+        --file secrets/rp-metrics-int.pem >/dev/null
+   az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name rp-mdsd \
+        --file secrets/rp-logging-int.pem >/dev/null
+   az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name cluster-mdsd \
+        --file secrets/cluster-logging-int.pem >/dev/null
+    az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name dev-arm \
+        --file secrets/arm.pem >/dev/null
+   az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name rp-firstparty \
+        --file secrets/firstparty.pem >/dev/null
+    az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-svc" \
+        --name rp-server \
+        --file secrets/localhost.pem >/dev/null
+    az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-por" \
+        --name portal-server \
+        --file secrets/localhost.pem >/dev/null
+    az keyvault certificate import \
+        --vault-name "$KEYVAULT_PREFIX-por" \
+        --name portal-client \
+        --file secrets/portal-client.pem >/dev/null
+    ```
+
+
+1. Mirror the OpenShift images to your new ACR
+    <!-- TODO (bv) allow mirroring through a pipeline would be faster and a nice to have -->
+
+    1. Setup mirroring environment variables
+        ```bash
+        export DST_ACR_NAME=${USER}aro
+        export SRC_AUTH_QUAY=FILL_IN # Get quay auth https://cloud.redhat.com/openshift/create/local -> Download Pull Secret
+        export SRC_AUTH_REDHAT=$(echo $USER_PULL_SECRET | jq -r '.auths."registry.redhat.io".auth')
+        export DST_AUTH=$(echo -n '00000000-0000-0000-0000-000000000000:'$(az acr login -n ${DST_ACR_NAME} --expose-token | jq -r .accessToken) | base64 -w0)
+
+    1. Login to the Azure Container Registry
+        ```bash
+        docker login -u 00000000-0000-0000-0000-000000000000 -p "$(echo $DST_AUTH | base64 -d | cut -d':' -f2)" "${DST_ACR_NAME}.azurecr.io"
+        ```
+
+    1. Run the mirroring
+        > The `latest` argument will take the InstallStream from `pkg/util/version/const.go` and mirror that version
+        ```bash
+        go run ./cmd/aro mirror latest
+        ```
+
+    1. Push the ARO image to your ACR
+        ```bash
+        make publish-image-aro-multistage
+        ```
+
+1. Delete the existing VMSS
+    > __NOTE:__ This needs to be deleted as deploying won't recreate the VMSS if the commit hash is the same.
+    ```bash
+    az vmss delete -g ${RESOURCEGROUP} --name rp-vmss-$(git rev-parse --short=7 HEAD)$([[ $(git status --porcelain) = "" ]] || echo -dirty)
```

Comment:
@jim-minter I've had to delete my VMSS before running `make deploy` a second time after all the images/secrets are loaded in due to it not redeploying the VMSS instance and continuously failing on pulling the ARO image, even when it's in the registry.

Is there logic we should change on deploying VMSS so that it will redeploy if the VMSS name is the same?

Reply:
MJ answered my question - deleting the VMSS here is valid and a good action to spell out; resolving...